casepro/cases/tests.py

```diff
@@ -1018,6 +1018,85 @@ def test_create_and_read(self):
         self.assertEqual(response.status_code, 302)
 
 
+class SystemUserTest(BaseCasesTest):
+    def setUp(self):
+        super(SystemUserTest, self).setUp()
+
+        self.ann = self.create_contact(self.unicef, 'C-001', "Ann")
+        d0 = datetime(2014, 1, 2, 12, 0, tzinfo=pytz.UTC)
+        msg1 = self.create_message(self.unicef, 101, self.ann, "Test Message", [self.aids], created_on=d0)
+        self.case = self.create_case(self.unicef, self.ann, self.moh, msg1)
+
+    def test_get_or_create_no_user(self):
+        # SystemUser.get_or_create() should return an existing instance or create a new one if none exists
+        self.assertEqual(SystemUser.objects.count(), 0)
+        user1 = SystemUser.get_or_create()
+        self.assertEqual(SystemUser.objects.count(), 1)
+        user2 = SystemUser.get_or_create()
+        self.assertEqual(SystemUser.objects.count(), 1)
+        self.assertEqual(user1, user2)
+
+    def test_add_note_as_system_user(self):
+        # SystemUsers should be able to access methods wrapped by the decorator
+        sysUser = SystemUser.get_or_create()
+        self.case.add_note(sysUser, "test note")
+
+        actions = CaseAction.objects.all()
+        self.assertEqual(len(actions), 1)
+        self.assertEqual(actions[0].created_by.pk, sysUser.pk)
+        self.assertEqual(actions[0].note, "test note")
+
+    def test_reassign_as_system_user(self):
+        # System reassigns case
+        self.assertEqual(self.case.assignee, self.moh)
+        partner = Partner.create(self.unicef, "Internal", False, [])
+        sysUser = SystemUser.get_or_create()
+        self.case.reassign(sysUser, partner)
+
+        actions = CaseAction.objects.all()
+        self.assertEqual(len(actions), 1)
+        self.assertEqual(actions[0].created_by.pk, sysUser.pk)
+        self.assertEqual(actions[0].assignee, partner)
+        self.assertEqual(actions[0].action, 'A')
+
+
+class CaseActionTest(BaseCasesTest):
```

Comment:
I'm a bit confused by these tests. It seems we didn't change anything in the PR related to case actions.